genai/internal/cmd/gen-examples/gen-examples.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This code generator takes examples from the internal/snippets directory
+// and copies them to "official" examples in genai/example_test.go, while
+// removing snippet comments (between [START...] and [END...]) that are used
+// for website documentation purposes.
+//
+// Run from the root directory:
+//
+//   go run ./genai/internal/cmd/gen-examples/ < genai/internal/snippets/example_test.go > genai/example_test.go
```

Comment:
Add a go:generate in the root to do this?

Reply:
What do you mean by "in the root"? The origin file? I was worried that it gets copied and then the destination file will have the same go:generate line which may run again
I suppose I can tweak the tool to remove the go:generate line when copying; what do you think?

Reply:
I wasn't clear. I was thinking genai/client.go.
Just so you could call `go generate` in the genai directory and everything would happen.
But maybe that's a bad idea. We don't necessarily want to run protoveneer every time we update the examples.
Could you put it in another .go file in the snippets dir that isn't copied?
I'm also fine just not doing it.

Reply:
I'm going to get this in for now, so we can point them to actual samples in the format they want and get the ball rolling. We can work out the exact mechanics of generation later - I'll think about it more and will send a follow-up